clowder-framework/clowder2-frontend
# Keycloak js

Use keycloak-js to handle login/logout and refresh token. **Note that "register" is not included!**

To test:
- Unzip and then import the keycloak setting:
[realm-export.json.zip](https://github.com/clowder-framework/clowder2-frontend/files/8474844/realm-export.json.zip). Double check if there is a "clowder2-frontend" client exist
- I moved the home routes to `/dashboard`; go to `/dashboard` should automatically redirect to keycloak for login
- Logout will not only wipe out the cookies but also log user out in keycloak. Please check the 
<img width="1402" alt="image" src="https://user-images.githubusercontent.com/13950475/163011038-586bbd07-cb31-472c-8efe-01ef048fef3c.png">
- Refresh seems to happen automatically with this keycloak wrapper. To mimic,  go to advance setting on the `clowder2-frontend` client, and toggle the token expiration time to 1 minute. Refresh on frontend /dashboard route after 1 minute and watch the URL change. 
<img width="692" alt="Screen Shot 2022-04-12 at 11 24 37 AM" src="https://user-images.githubusercontent.com/13950475/163011456-e31d01a0-883f-4e08-8f5b-8b7b6df92dfa.png">
- To be safe, also manually implemented the refresh token process once backend return 401 error.
